### ISSUE TITLE
Support for nullable optional parameters

### ISSUE THREAD

**Issue by payam-zahedi** [6thsolution/dart_sealed#4]
After working with this package for a while, I have noticed that the package doesn't support for optional parameters. 
I think users will have a better experience if they do not have to submit optional parameters.

**Comment by FatulM** [comment]
HI,
One way to implement this is to omit `required` from nullable arguments in constructors.
What's your opinion ?
@SaeedMasoumi

**Comment by SaeedMasoumi** [comment]
The generated file will be like from
```
 const factory State.success({
    required A a,
    required B? b,
  }) = StateSuccess;
```
to 
```
 const factory State.success({
    required A a,
    B? b,
  }) = StateSuccess;
```
?
It makes sense to be like the second form

**Comment by payam-zahedi** [comment]
> ```
>  const factory State.success({
>     required A a,
>     B? b,
>   }) = StateSuccess;
> ```
> 
 Yes, that make sense and has better experience.

**Comment by payam-zahedi** [comment]
Does the package also support Optional positional parameters? Something like this?
```
 void success(bool status, [ String? value]);
```

**Comment by FatulM** [comment]
So, I'll change implementation ASAP
😊

**Comment by FatulM** [comment]
> Does the package also support Optional positional parameters? Something like this?
> ```
>  void success(bool status, [ String? value]);
> ```

No,
All of parameters are named ...
Is this feature needed ?

**Comment by payam-zahedi** [comment]
It's not something I need, but others may need it.

**Comment by FatulM** [comment]
Hi,
I've published version 1.9.0 which addresses your issue ...

**Comment by payam-zahedi** [comment]
That's awesome @FatulM, we can close this issue now.
